api/api_models/photos.js

```diff
@@ -0,0 +1,82 @@
+import { BaseModel, string, date } from "./base_model.js";
+import Photography_status from "../models/enums/photography_status.js";
+import { ErrorInternalAPIModelValidation } from "../error/internal_errors.js";
+
+/**
+ * Represents the http request body required
+ * to create a photo in the database.
+ */
+export class PhotoCreate extends BaseModel{
+    static schema = {
+        url: { type: String, unique: true, required: true },
+        tags: [{ type: Schema.Types.ObjectId, ref: "PhotoTags" }],
```

Comment:
Note that API models are separate from Mongoose Schema models. They are defined syntactically differently. API models define their own supported types, and supported options. `unique` and `ref` are not valid options in API models, since they have nothing to do with database - it's just JSON validation.

Use API model syntax to denote lists, and datatypes.
```js
{
  tags: { type: [object]}
}
```